robotbase/src/main/java/com/team2357/frc2022/commands/ExampleTrajectoryCommand.java

```diff
@@ -0,0 +1,80 @@
+import edu.wpi.first.math.trajectory.TrajectoryConfig;
+import edu.wpi.first.math.trajectory.TrajectoryGenerator;
+import edu.wpi.first.math.trajectory.constraint.DifferentialDriveVoltageConstraint;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.RamseteCommand;
+
+public class ExampleTrajectoryCommand extends CommandLoggerBase{
+    FalconTrajectoryDriveSubsystem m_driveSub;
+
+    public ExampleTrajectoryCommand(FalconTrajectoryDriveSubsystem driveSub) {
+        m_driveSub = driveSub;
+    }
+
+    public Command getRamsete() {
+        var autoVoltageConstraint = new DifferentialDriveVoltageConstraint(
+            new SimpleMotorFeedforward(
+                Constants.DRIVE.KS_VOLTS,
+                Constants.DRIVE.KV_VOLTS_SECONDS_PER_METER,
+                Constants.DRIVE.KA_VOLTS_SECONDS_SQUARED_PER_METER),
+            Constants.DRIVE.DRIVE_KINEMATICS,
+            Constants.DRIVE.MAX_VOLTAGE);
+
+        // Create config for trajectory
+        TrajectoryConfig config = new TrajectoryConfig(
+            Constants.DRIVE.MAX_SPEED_METERS_PER_SECOND,
+            Constants.DRIVE.MAX_ACCELERATION_METERS_PER_SECOND_SQUARED)
+                // Add kinematics to ensure max speed is actually obeyed
+                .setKinematics(Constants.DRIVE.DRIVE_KINEMATICS)
+                // Apply the voltage constraint
+                .addConstraint(autoVoltageConstraint);
+
+        // An example trajectory to follow. All units in meters.
+        Trajectory exampleTrajectory = TrajectoryGenerator.generateTrajectory(
+            // Start at the origin facing the +X direction
+            new Pose2d(0, 0, new Rotation2d(0)),
+            // Pass through these two interior waypoints, making an 's' curve path
+            List.of(new Translation2d(1, 0), new Translation2d(2, 0)),
+            // End 3 meters straight ahead of where we started, facing forward
+            new Pose2d(3, 0, new Rotation2d(0)),
+            // Pass config
+            config);
+
+        RamseteCommand ramseteCommand = new RamseteCommand(
+            exampleTrajectory,
+            m_driveSub::getPose,
+            new RamseteController(Constants.DRIVE.RAMSETE_B, Constants.DRIVE.RAMSETE_ZETA),
+            new SimpleMotorFeedforward(
+                Constants.DRIVE.KS_VOLTS,
+                Constants.DRIVE.KV_VOLTS_SECONDS_PER_METER,
+                Constants.DRIVE.KA_VOLTS_SECONDS_SQUARED_PER_METER),
+            Constants.DRIVE.DRIVE_KINEMATICS,
+            m_driveSub::getWheelSpeeds,
+            new PIDController(Constants.DRIVE.P_DRIVE_VEL, 0, 0),
+            new PIDController(Constants.DRIVE.P_DRIVE_VEL, 0, 0),
+            // RamseteCommand passes volts to the callback
+            m_driveSub::setTankDriveVolts,
+            m_driveSub);
+
+        // Reset odometry to the starting pose of the trajectory.
+        m_driveSub.resetOdometry(exampleTrajectory.getInitialPose());
```

Comment:
I don't think it makes sense for what looks like a "getter" to do mutating calls here and then return a command.

Given the above comment I already placed, this is my suggestion (but let me know if I'm missing something!):

1. Make this a base class (maybe rename this to `TrajectoryCommand`)
2. Add a `calculateTrajectory()` function designed to be overridden by "real" commands. It should return a `Trajectory` object.
3. Put all the rest of this in the `start()` function of the base TrajectoryCommand class, and don't return the command, just run it.